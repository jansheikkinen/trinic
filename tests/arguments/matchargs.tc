// Match Arguments

function main(void) -> void do
  match a do
    1    => 2;
    3, 4 => 5;
  end
end
