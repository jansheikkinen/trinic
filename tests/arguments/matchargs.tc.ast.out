
### AST GENERATION ###
Declaration FUNCTION  main
  Stmt MATCH
    Expr VARIABLE a
    Arg MATCH_ARM
      Match_Arm
        Arg EXPR
          Expr LITERAL
            INT 1
        Stmt EXPRESSION
          Expr LITERAL
            INT 2
      Match_Arm
        Arg EXPR
          Expr LITERAL
            INT 3
          Expr LITERAL
            INT 4
        Stmt EXPRESSION
          Expr LITERAL
            INT 5
